### ISSUE TITLE
Exception is raised while loading the fine tuned model

### ISSUE THREAD

**Issue by himanshucapoor** [vllm-project/vllm#455]
Hi,

I have fine-tuned the `mpt-30b` and wanted to load it using vllm for inference

`model = LLM(model=/local_path, tensor_parallel_size=self.vllm_args.tensor_parallel_size,
                             pipeline_parallel_size=self.vllm_args.pipeline_parallel_size,
                             gpu_memory_utilization=self.vllm_args.gpu_memory_utilization)`

I am getting the following error:

`/local_path does not appear to have a file named mosaicml/mpt-30b--configuration_mpt.py. Checkout 'https://huggingface.co//home/ubuntu/fine_tune/op/mpt_30b_gradient_accumu_checkpoint_11072023/None' for available files.`

Wanted to check how to load my locally fine tuned model in the vllm

**Comment by himanshucapoor** [comment]
Along with the saving the fine-tuned model using `trainer.save_model`, we need to save the configuration also in the local directory to solve the above error. ```
name = 'mosaicml/mpt-30b'
config = transformers.AutoConfig.from_pretrained(name, trust_remote_code=True)
config.save_pretrained("/local_path")
```

solved the above issue, however when i am trying to load the model using the below code in jupyter notebook

`model = LLM(model=/local_path, tensor_parallel_size=self.vllm_args.tensor_parallel_size, pipeline_parallel_size=self.vllm_args.pipeline_parallel_size, gpu_memory_utilization=self.vllm_args.gpu_memory_utilization)
`
i am getting this in the logs
`Started a local Ray instance.`
and then error on the jupyter notebook The kernel appears to have died. It will restart automatically.

**Comment by himanshucapoor** [comment]
in the locally saved model directory using `trainer.save_model` it's creating the `training_args.bin` file also and in `weight_utils.py` in the method `load_tensor_parallel_weights` on line 77 there's a code

`for bin_file in hf_bin_files:
            state = torch.load(bin_file, map_location="cpu")
            for name, param in state.items():
                yield name, param`

where code is iterating over the bin files. So when it will be de-serialised to  `TrainingArguments` class object it will not have `items` attribute leading to above error. Should i raise a PR for the fix ?

**Comment by WoosukKwon** [comment]
Hi @himanshucapoor thanks for raising the issue. I think it's fixed by #373  in the current version of `main`?
https://github.com/vllm-project/vllm/blob/c894836108732d0cbb6fce15aeda8de1218a380d/vllm/model_executor/weight_utils.py#L42-L45

**Comment by himanshucapoor** [comment]
Hi @WoosukKwon yes its the same issue. Closing the issue in this case. For now i have manually deletes the `training_args.bin` from my directory and its working fine. When is this fix scheduled for rleease ?
